website/blog/2024-02-28-7.24.0.md

```diff
@@ -0,0 +1,91 @@
+---
+layout: post
+title:  "7.24.0 Released: Decorator updates and JSON modules imports"
+author: Babel Team
+authorURL: https://twitter.com/babeljs
+date:   2024-02-28 0:00:00
+categories: announcements
+share_text: "Babel 7.24.0 Released"
+---
+
+Babel 7.24.0 is out!
+
+We updated our [Decorators](https://github.com/tc39/proposal-decorators/) implementation to match the latest version of the proposal, and we improved the way we transform class private fields and methods. We also added support for importing [JSON modules](https://github.com/tc39/proposal-json-modules) in browsers and Node.js, an older Stage 3 proposal depending on [Import Attributes](https://github.com/tc39/proposal-import-attributes).
+
+You can read the whole changelog [on GitHub](https://github.com/babel/babel/releases/tag/v7.23.0).
+
+<!-- truncate -->
+
+If you or your company want to support Babel and the evolution of JavaScript, but aren't sure how, you can donate to us on our [Open Collective](https://github.com/babel/babel?sponsor=1) and, better yet, work with us on the implementation of [new ECMAScript proposals](https://github.com/babel/proposals) directly! As a volunteer-driven project, we rely on the community's support to fund our efforts in supporting the wide range of JavaScript users. Reach out at [team@babeljs.io](mailto:team@babeljs.io) if you'd like to discuss more!
+
+## Highlights
+
+### Decorators updates ([#16242](https://github.com/babel/babel/pull/16242))
+
+The [decorators](https://github.com/tc39/proposal-decorator-metadata/) Stage 3 proposals received many minor updates in the past years, and unfortunately all the implementations in the various tools implement [slightly different versions](https://github.com/evanw/decorator-tests) of it.
+
+We updated our implementation to match the latest version of the proposal. This version has also been implemented in TypeScript and is currently being implemented natively in browsers. The main difference compared to the version previously implemented by Babel is in the execution order of initializers registered through the `context.addInitializer` method.
+
+You can enable this version of the proposal by setting the `"version": "2023-11"` option in `@babel/plugin-proposal-decorators`:
+```json
+{
+    "plugins": [
+        ["@babel/plugin-proposal-decorators", {
+            "version": "2023-11"
+        }]
+    ]
+}
+```
```

Comment:
We can add an REPL link, the link will work when we publish 7.24.0 and merge https://github.com/babel/website/pull/2860
```
You can also try the new decorators proposal in the online Babel [REPL](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=GYVwdgxgLglg9mABAGzgczQUwE4AoCUiA3gFCKID0FiA8gA6ZICecI2A5AM6IBG2cAd045EEBJzjJMAGjKVqAQzAATRIwU8piAEQBRAG4LkIBVEzbEMJFAAWmRJxjLMPBdgCEcsWAlSAdKhouG5oIAC2jFCc-ADcJAC-JCQQyAqc3ACCxHIAAoFY2HIRtnDKBMSJ8UA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=&version=7.24.0&externalPlugins=%40babel%2Fplugin-proposal-decorators%407.24.0&assumptions=%7B%7D).
```

Reply:
https://github.com/babel/website/pull/2873